packages/core-modules/contracts/modules/CoreElectionModule.sol

```diff
@@ -152,24 +157,30 @@ contract CoreElectionModule is IElectionModule, ElectionStorage, OwnableMixin {
 
         uint votePower = ERC20(store.electionTokenAddress).balanceOf(msg.sender);
 
-        for (uint i = 0; i < candidates.length; i++) {
-            address candidate = candidates[i];
+        address[] memory candidates = new address[](sortedCandidates.length);
 
-            if (electionData.nomineeIndexes[candidate] == 0) {
-                revert InvalidCandidate(candidate);
+        for (uint i = 0; i < sortedCandidates.length; i++) {
+            address candidate = sortedCandidates[i];
+            uint position = positions[i];
+
+            // Validate positions repetition
+            if (candidates[position] != address(0)) {
+                revert InvalidCandidatesPositions();
             }
 
-            // Check that all the values on the candidates Array are unique
-            // TODO: Remove the nested loops https://github.com/Synthetixio/synthetix-v3/issues/518
-            if (i < candidates.length - 1) {
-                for (uint256 j = i + 1; j < candidates.length; j++) {
-                    address nextCandidate = candidates[j];
-                    if (candidate == nextCandidate) {
-                        revert InvalidCandidateRepeat(candidate);
-                    }
+            // Validate candidates repetition
+            if (i > 0) {
+                address prev = sortedCandidates[i - 1];
+                if (candidate <= prev) {
+                    revert InvalidCandidateRepeat(candidate);
```

Comment:
Maybe name this one `DupicateCandidate`?

Reply:
Done! to `DuplicateCandidate`